extensions/amp-analytics/0.1/linker-manager.js

```diff
@@ -308,19 +304,43 @@ export class LinkerManager {
 
 
   /**
-   * Add the linker pairs as <input> elements to form.
+   * Add the linker data to form. If action-xhr is present we can set a new
+   * linker-xhr to use, if not we fallback to adding hidden inputs.
    * @param {!Element} form
    * @param {string} linkerName
    */
   addDataToForm_(form, linkerName) {
-    if (!this.resolvedLinkers_[linkerName]) {
+    const linkerValue = this.resolvedLinkers_[linkerName];
+    if (!linkerValue) {
       return;
     }
 
+    // Runtime controls submits with `action-xhr`, so we can point it at the new
+    // url set as linker-xhr attribute.
+    const actionXhrUrl = form.getAttribute('action-xhr');
+    if (actionXhrUrl) {
+      form.setAttribute('linker-xhr',
```

Comment:
As discussed offline. My concern is that there's might be a privacy issue since cid can be written to DOM. Especially that I remembered we allow  amp-analytics to pass html by selector.

Reply:
Agreed. Setter also solved this problem.